File: python/n350_InterB.py
# https://leetcode-cn.com/problems/intersection-of-two-arrays-ii/
class Solution:
    def intersect(self, nums1, nums2):
        s1 = sorted(nums1)
        s2 = sorted(nums2)
        i = j = 0
        res = []
        while i < len(s1) and j < len(s2):
            if s1[i] == s2[j]:
                res.append(s1[i])
                i += 1
                j += 1
            elif s1[i] > s2[j]:
                j += 1
            else:
                i += 1
        return res
        
        
if __name__ == "__main__":
    nums1 = [1, 2, 2, 1]
    nums2 = [2, 2]
    solu = Solution()
    print(solu.intersect(nums1, nums2))
        

File: cpp/n350_interB.cpp
// https://leetcode-cn.com/problems/intersection-of-two-arrays-ii/submissions/
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
#include <algorithm>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        sort (nums1.begin(), nums1.end());
        sort (nums2.begin(), nums2.end());
        vector<int> res;
        size_t i = 0, j = 0;
        while (i < nums1.size() && j < nums2.size()){
            if (nums1[i] == nums2[j]){
                res.push_back(nums1[i]);
                ++i;
                ++j;
            }
            else if (nums1[i] < nums2[j]) ++i;
            else ++j;
        }
        return res;
    }
};

int main(){
    return 0;
}